File: app/src/main/java/com/designthinkingpractical/app/domain/model/BloodBank.kt
package com.designthinkingpractical.app.domain.model

data class BloodBank(
    val id: String,
    val name: String,
    val address: String,
    val phone: String,
    val inventory: List<BloodInventory>
)

data class BloodInventory(
    val bloodGroup: String,
    val units: Int
)

File: backend/models/BloodBank.js
const mongoose = require('mongoose');

const bloodBankSchema = new mongoose.Schema({
  name: { type: String, required: true },
  address: { type: String, required: true },
  phone: { type: String, required: true },
  location: {
    lat: { type: Number, required: true },
    lng: { type: Number, required: true }
  },
  inventory: [{
    bloodGroup: { type: String, required: true },
    units: { type: Number, default: 0 }
  }]
}, { timestamps: true });

module.exports = mongoose.model('BloodBank', bloodBankSchema);
